docs/manual/returns-receiver-checker.tex

```diff
@@ -0,0 +1,118 @@
+\htmlhr
+\chapterAndLabel{Returns Receiver Checker}{returns-receiver-checker}
+
+The Returns Receiver Checker enables documenting and checking that a method
+returns its receiver (i.e., the \<this> parameter).
+
+There are two ways to run the Returns Receiver Checker.
+\begin{itemize}
+\item
+Typically, it is automatically run by another checker.
+\item
+Alternately, you can run just the Returns Receiver Checker, by
+supplying the following command-line options to javac:
+\code{-processor org.checkerframework.common.returnsreceiver.ReturnsReceiverChecker}
+\end{itemize}
+
+
+\sectionAndLabel{Annotations}{returns-receiver-checker-annotations}
+
+The qualifier \refqualclass{common/returnsreceiver/qual}{This} on the return
+type of a method indicates that the method returns its receiver.  Methods
+that return their receiver are common in so-called ``fluent'' APIs.  Here
+is an example:
+
+\begin{Verbatim}
+class MyBuilder {
+  @This MyBuilder setName(String name) {
+    this.name = name;
+    return this;
+  }
+}
+\end{Verbatim}
+
+An \refqualclass{common/returnsreceiver/qual}{This} annotation can only be
+written on the return type of a method or in a downcast.
```

Comment:
Regarding your first comment:
I see no harm to writing it explicitly on the receiver type.  I also see no benefit.

Some advantages of the current choice:
 * It is marginally less verbose.
 * Permitting only one syntax will avoid confusion from readers seeing different variants.

Regarding your second comment:  good point, @msridhar should address that.

Reply:
> Should this mention that the return type has to be exactly the type of the enclosing method? (Or is there a different rule?)

I'm not sure I understand.  Did you mean enclosing class?  If so, that is not true; the return type, e.g., can be a superclass.  This is what occurs in the generated sub-class for an AutoValue Builder (see the `fig-autovalue-builder-generated` below).  If you meant something else, please let me know.

Reply:
Ah, yes, the enclosing class or superclasses makes sense. Does `@This Object` always make sense? Whatever the rules are, it would be nice to document them.

Reply:
I think `@This Object` makes perfect sense, though I don't immediately have a concrete use case.  Are you sure it would be useful to document these constraints on the return type?  They are directly implied by the fact that the checker ensures the method returns `this`.  I am trying to write a sentence or two on it but it does not seem useful.

Reply:
This was the first I read about this type system, so I was wondering when I can use it.
I find descriptions of the rules useful in documentation.
If you prefer being vague, that's fine with me.

Reply:
> Shouldn't it be legal to explicitly write it on the receiver type?

It is now legal to write on the receiver type.